harness/tests/experiment/pytorch/test_deepspeed_autotuning.py

```diff
@@ -0,0 +1,206 @@
+    SimpleBatchSearchMethod,
+    DSATRandomSearchMethod,
+)
+
+
+BASE_EXPERIMENT_FIXTURE_PATH = (
+    pathlib.Path(__file__).resolve().parent.parent.joinpath("fixtures/deepspeed_autotune")
+)
+
+MODEL_INFO_PROFILE_METRIC_FIXTURE = {
+    "num_params": 60192808,
+    "trainable_num_params": 60192808,
+    "activation_mem_per_gpu": 89828352,
+    "rank": 0,
+    "gpu_mem": 15843721216,
+}
+
+
+@mock.patch("determined.experimental.client.create_experiment")
+def test_autotuning_module_transforms(
+    create_experiment_mock: mock.MagicMock,
+) -> None:
+    model_dir = BASE_EXPERIMENT_FIXTURE_PATH.joinpath("example_experiment")
+    config_path = model_dir.joinpath("autotune_config.yaml")
+    args = Namespace(
+        config_path=config_path,
+        model_dir=model_dir,
+        tuner_type="random",
+        search_runner_config=None,
+    )
+    autotune.run_autotuning(args)
+    create_experiment_mock.assert_called_once()
+    submitted_single_searcher_config_dict = create_experiment_mock.call_args_list[0].kwargs[
+        "config"
+    ]
+    # The Search Runner should be defaulted to a "single" searcher
+    assert submitted_single_searcher_config_dict.get("searcher", {}).get("name", "") == "single"
+
+
+def test_simple_search_method_happy_path() -> None:
+    model_dir = BASE_EXPERIMENT_FIXTURE_PATH.joinpath("example_experiment")
+    config_path = model_dir.joinpath("autotune_config.yaml")
+    submitted_config_dict = _utils.get_dict_from_yaml_or_json_path(config_path)
+
+    with tempfile.TemporaryDirectory() as searcher_dir:
+        searcher_dir = pathlib.Path(searcher_dir)
+        search_method = SimpleBatchSearchMethod(
+            submitted_config_dict=submitted_config_dict,
+            model_dir=model_dir,
+        )
+        mock_master_obj = MockMaster(
+            all_metrics=[MODEL_INFO_PROFILE_METRIC_FIXTURE, {"throughput": 1.0}]
+        )
+        search_runner = MockMasterSearchRunner(search_method, mock_master_obj, searcher_dir)
+        search_runner.run(exp_config={}, context_dir="", includes=None)
+
+    # TODO: Use a more dynamic value if/when we enable users to configure this
+    exp_num_trials = _defaults.AUTOTUNING_DICT["tuner_num_trials"]
+    assert len(search_runner.state.trials_created) == exp_num_trials
+    assert len(search_runner.state.trials_closed) == exp_num_trials
+    assert len(search_runner.state.trial_progress) == exp_num_trials
+    # TODO: Handle the progress being 6 every time...
+    # for trial_uuid in search_runner.state.trial_progress:
+    #     assert(search_runner.state.trial_progress[trial_uuid] == 1.0)
+
+
+# TODO: Check that the trial tracker stops early when we trigger the early stopping criteria
+
+
+def test_random_search_method() -> None:
+    pass
+
+
+def test_trial_tracker() -> None:
+    pass
+
+
+def test_mbs_binary_search() -> None:
+    pass
+
+
+class MockMaster:
+    def __init__(self, all_metrics: List[Union[float, Dict[str, Any]]]) -> None:
```

Comment:
I'm a little confused about how `all_metrics` is being used here and in the tests below.  Seems like all of the metrics are being pre-determined before the actual experiments run?

Reply:
`all_metrics` is the sequential list of metrics that the MockedMaster will return. It will continuously send the last one until the test is over.